twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -296,62 +299,94 @@ public function verifyChallenge(IUser $user, $challenge)
      * @throws TwoFactorException
      */
     public function authenticate($username, $password) {
-        // Read config
-        $url = $this->getBaseUrl() . "validate/check";
-        $realm = $this->getAppValue('realm', '');
         $error_message = "";
+
+        // Read config
         $options = $this->getClientOptions();
-        $options['body'] = ['user' => $username,
-            'pass' => $password,
-            'realm' => $realm];
-        // The verifyChallenge is called with additional parameters in case of challenge response:
-        $transaction_id = $this->session->get("pi_transaction_id");
-        $signatureData = $this->request->getParam("signatureData");
-        $clientData = $this->request->getParam("clientData");
-        $this->log("debug", "transaction_id: " . $transaction_id);
-        $this->log("debug", "signatureData: " . $signatureData);
-        $this->log("debug", "clientData: " . $clientData);
-
-        if ($transaction_id) {
-            // add transaction ID in case of challenge response
-            $options['body']["transaction_id"] = $transaction_id;
+
+        $pushResponse = $this->session->get("pi_PUSH_Response");
+        if ($password && $this->session->get("pi_hideOTPField") === false) {
```

Comment:
Right now you cannot use u2f, if a push token is available. So we should edit this a bit like `if ($password || $signatureData)`. So we need to read this earlier, I think (see line 325).
Do we need the check for `pi_hideOTPField` here?

Reply:
Fixed.
No we don't need it.